src/analysis/raster/qgsrastercalculator.cpp

```diff
@@ -735,7 +735,7 @@ QVector<QgsRasterCalculatorEntry> QgsRasterCalculatorEntry::rasterEntries()
   for ( ; layerIt != layers.constEnd(); ++layerIt )
   {
     QgsRasterLayer *rlayer = qobject_cast<QgsRasterLayer *>( layerIt.value() );
-    if ( rlayer && rlayer->dataProvider() && rlayer->providerType() == QLatin1String( "gdal" ) )
+    if ( rlayer && rlayer->dataProvider() && ( rlayer->providerType() == QLatin1String( "gdal" ) || rlayer->providerType() == QLatin1String( "postgresraster" ) ) )
```

Comment:
is there a reason to test the providerType() at all ?

Reply:
I think so @rouault, at least for now.

If not, `wms` layers, for example, would also be listed in the raster calculator. Are we able to handle `wms` layers properly in the raster calculator?

I've done a quick test. I was not able to use the raster calculator with a WMS layer. Raster Calculator is not able to estimate the row and columns numbers from the source (using the Selected Layer Extent button), for example, as we do for a `gdal` or `postgresqraster` raster. Writing down the size of the output raster was not enough. I've got many `src/core/raster/qgsrasterblock.h:818 : (valueAndNoData) [0ms] Data block not allocated` errors while reading the data blocks.

In my opinion, we would need to review all descendant classes of `QgsRasterLayer` before allowing them to be listed in the raster calculator.

What do you think? Should we list all `QgsRasterLayer` descendant classes here and fix the failing issues?

Reply:
perhaps testing for (capabilities() & QgsRasterDataProvider::Size) != 0 which is true for the GDAL provider and PostgisRaster one, but not the WMS one

Reply:
I've improved the condition. It doesn't use provider literal names anymore. Thanks @rouault for the tip.

I've also removed the condition from the dialog itself. The list is provided by `rasterEntries()` where the condition makes sense. It was unnecessary duplicated on the dialog side.